### PR TITLE
fix CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by AFMahmuda**
fix https://github.com/TheAlgorithms/Go/issues/513

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,16 +1,21 @@
...
 
-## Before contributing
+Before contributing
+------
 
 Welcome to [TheAlgorithms/Go](https://github.com/TheAlgorithms/Go)! Before submitting pull requests, please make sure that you have **read the whole guidelines**. If you have any doubts about this contribution guide, please open [an issue](https://github.com/TheAlgorithms/Go/issues/new/choose) or ask in our [Discord server](https://discord.gg/c7MnfGFGa6) / [Gitter](https://gitter.im/TheAlgorithms), and clearly state your concerns.
 
-## Contributing
+Contributing
+------
```
Comment (siriak) [inline CONTRIBUTING.md:10]: Please use # for headings, it's clearer IMO

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -19,13 +24,18 @@ Being a contributor at The Algorithms, we request you to follow the points menti
...
 
-**Issues** Please avoid opening issues asking to be "assigned” to a particular algorithm. This merely creates unnecessary noise for maintainers. Instead, please submit your implementation in a pull request, and it will be evaluated by project maintainers.
+- You can add new algorithms or data structures that are **not present in the repository** or that can **improve** the old implementations (**documentation**, **improving test cases**, removing bugs, or in any other reasonable sense)
+
+#### Issues
+
+- Please avoid opening issues asking to be "assigned” to a particular algorithm. This merely creates unnecessary noise for maintainers. Instead, please submit your implementation in a pull request, and it will be evaluated by project maintainers.
 
 ### Making Changes
+---
```
Comment (siriak) [inline CONTRIBUTING.md:38]: GitHub adds horizontal lines here, but it's supposed to be a level 2 heading

**Review by siriak — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md (hunk 1 above)
Comment (AFMahmuda) [reply CONTRIBUTING.md:10]: agree. but i want to keep the lines for clearer separation.

File: CONTRIBUTING.md (hunk 2 above)
Comment (AFMahmuda) [reply CONTRIBUTING.md:38]: this is intentional, i personally find it easier to read the document this way. if you disagree, then I can remove it.

File: CONTRIBUTING.md (hunk 1 above)
Comment (AFMahmuda) [reply CONTRIBUTING.md:10]: level 3 and 4 heading is not obvious by default.

Comment (AFMahmuda) [reply CONTRIBUTING.md:10]: adjusted

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -1,16 +1,21 @@
 # CONTRIBUTION GUIDELINES
+======
```
Comment (tjgurwara99) [inline CONTRIBUTING.md:2]: I don't think we need this one. The GitHub markdown automatically adds one on the top heading (though it is not in the spec - afaik). If you look at the Contributing page, it is present. 

In fact, from what I can see, there is an actual issue on your fork [page](https://github.com/AFMahmuda/Go/blob/docs/CONTRIBUTING.md)

**Review by tjgurwara99 — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -1,16 +1,21 @@
...
 
 ## Before contributing
+------
```
Comment (tjgurwara99) [inline CONTRIBUTING.md:5]: Not needed - GitHub automatically adds that in h1, and h2 headings.

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -1,16 +1,21 @@
...
 
 Welcome to [TheAlgorithms/Go](https://github.com/TheAlgorithms/Go)! Before submitting pull requests, please make sure that you have **read the whole guidelines**. If you have any doubts about this contribution guide, please open [an issue](https://github.com/TheAlgorithms/Go/issues/new/choose) or ask in our [Discord server](https://discord.gg/c7MnfGFGa6) / [Gitter](https://gitter.im/TheAlgorithms), and clearly state your concerns.
 
 ## Contributing
+------
```
Comment (tjgurwara99) [inline CONTRIBUTING.md:10]: Not needed here either

**Review by tjgurwara99 — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md (hunk 4 above)
Comment (AFMahmuda) [reply CONTRIBUTING.md:5]: thank you for pointing it out. will remove.

File: CONTRIBUTING.md (hunk 3 above)
Comment (AFMahmuda) [reply CONTRIBUTING.md]: adjusted

File: CONTRIBUTING.md (hunk 4 above)
Comment (AFMahmuda) [reply CONTRIBUTING.md]: adjusted

File: CONTRIBUTING.md (hunk 5 above)
Comment (AFMahmuda) [reply CONTRIBUTING.md]: adjusted, @tjgurwara99

**Review by siriak — APPROVED** [review]

**Review by tjgurwara99 — APPROVED** [review]

**Pull request merged**